File: swagger/src/main/java/com/example/swagger/domain/feed/presentation/FeedController.java
package com.example.swagger.domain.feed.presentation;

import com.example.swagger.domain.feed.presentation.dto.request.CreateFeedRequest;
import com.example.swagger.domain.feed.service.CreateFeedService;
import com.example.swagger.domain.feed.service.DeleteFeedService;
import com.example.swagger.domain.feed.service.UpdateFeedService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.Parameter;
import io.swagger.v3.oas.annotations.media.Content;
import io.swagger.v3.oas.annotations.media.Schema;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.responses.ApiResponses;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;

@Tag(name = "게시글", description = "게시글 관련 API 입니다.")
@RequiredArgsConstructor
@RequestMapping("/feeds")
@RestController
public class FeedController {

    private final CreateFeedService createFeedService;
    private final UpdateFeedService updateFeedService;
    private final DeleteFeedService deleteFeedService;

    @ResponseStatus(HttpStatus.CREATED)
    @PostMapping
    @Operation(summary = "게시글 생성하기")
    @ApiResponses(value = {
            @ApiResponse(responseCode = "201", description = "게시글 생성 성공",
                    content = @Content(schema = @Schema(hidden = true)))
    })
    public void createFeed(@RequestBody @Valid CreateFeedRequest request) {
        createFeedService.execute(request);
    }

    @ResponseStatus(HttpStatus.NO_CONTENT)
    @PatchMapping("/{feed-id}")
    @Operation(summary = "게시글 수정하기")
    @ApiResponse(responseCode = "204", description = "게시글 수정 성공",
            content = @Content(schema = @Schema(hidden = true)))
    public void updateFeed(@Parameter(description = "게시글 id") @PathVariable(name = "feed-id") Integer feedId,
            @RequestBody @Valid CreateFeedRequest request) {
        updateFeedService.execute(feedId, request);
    }

    @ResponseStatus(HttpStatus.NO_CONTENT)
    @DeleteMapping("/{feed-id}")
    @Operation(summary = "게시글 삭제하기")
    @ApiResponse(responseCode = "204", description = "게시글 삭제 성공",
            content = @Content(schema = @Schema(hidden = true)))
    public void deleteFeed(@Parameter(description = "게시글 id") @PathVariable(name = "feed-id") Integer feedId) {
        deleteFeedService.execute(feedId);
    }

}

File: practice-kotlin/src/main/kotlin/com/example/practicekotlin/domain/feed/presentation/FeedController.kt
package com.example.practicekotlin.domain.feed.presentation

import com.example.practicekotlin.domain.feed.presentation.dto.request.CreateFeedRequest
import com.example.practicekotlin.domain.feed.presentation.dto.request.UpdateFeedRequest
import com.example.practicekotlin.domain.feed.presentation.dto.response.QueryFeedListResponse
import com.example.practicekotlin.domain.feed.presentation.dto.response.QueryFeedResponse
import com.example.practicekotlin.domain.feed.service.*
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RequestMapping("/feeds")
@RestController
class FeedController(
    private val createFeedService: CreateFeedService,
    private val updateFeedService: UpdateFeedService,
    private val deleteFeedService: DeleteFeedService,
    private val queryFeedDetailService: QueryFeedDetailService,
    private val queryFeedListService: QueryFeedListService
) {

    @ResponseStatus(HttpStatus.CREATED)
    @PostMapping
    fun createFeed(@RequestBody request: CreateFeedRequest) {
        createFeedService.execute(request)
    }

    @ResponseStatus(HttpStatus.NO_CONTENT)
    @PatchMapping("/{feed-id}")
    fun updateFeed(@PathVariable("feed-id") feedId: Long,
                   @RequestBody request: UpdateFeedRequest) {
        updateFeedService.execute(feedId, request)
    }

    @ResponseStatus(HttpStatus.NO_CONTENT)
    @DeleteMapping("/{feed-id}")
    fun deleteFeed(@PathVariable("feed-id") feedId: Long) {
        deleteFeedService.execute(feedId)
    }

    @GetMapping("/{feed-id}")
    fun getFeedDetail(@PathVariable("feed-id") feedId: Long): QueryFeedResponse {
        return queryFeedDetailService.execute(feedId)
    }

    @GetMapping
    fun getFeedList(): QueryFeedListResponse {
        return queryFeedListService.execute()
    }

}